viral/offline_sequence_detection.py

```
@@ -0,0 +1,434 @@
+    if np.isnan(Pr).any():
+        raise ValueError("NaN values in Pr")
+
+    # and only bins with non-zero firing rates were used for offline Bayesian decoding. Posterior probabilities were subsequently normalized to one:
+
+    return Pr, pr_max
+    # TODO: check the matlab implementation for same result
+    # TODO: it is the same!!!
+
+
+def shuffle_pse_event(
+    posterior_probability_matrix: np.ndarray, n_shuffles: int = 5
```

Ruff (0.14.1)

Unused function argument: `n_shuffles`

(ARG001)